lib/rake/file_task.rb

```diff
@@ -14,14 +14,18 @@ class FileTask < Task
     # Is this file task needed?  Yes if it doesn't exist, or if its time stamp
     # is out of date.
     def needed?
-      !File.exist?(name) || out_of_date?(timestamp) || @application.options.build_all
+      begin
+        out_of_date?(File.mtime(name)) || @application.options.build_all
```

Comment:
Calling `File.mtime` here directly instead of `timestamp` because checking to see if it returned `Rake::LATE` was significantly slower. This is functionally the same as before as `out_of_date?` wouldn’t receive `Rake::LATE` except if there was a race condition between the first and second `File.exist` check.